### PR TITLE
cilium: Add empty types for TLS wrapper config to keep Envoy happy

### PR DESCRIPTION
**Pull request by jrajahalme**
Add empty types for ::cilium::UpstreamTlsWrapperContext and ::cilium::DownstreamTlsWrapperContext to avoid Envoy complaining about double registration. TLS wrapper does not need any config, so using the actual Envoy TLS config types was not necessary.

Signed-off-by: Jarno Rajahalme <jarno@isovalent.com>

### CODE REVIEW

File: tests/cilium_tls_http_integration_test.cc [hunk 1]
```diff
@@ -125,6 +126,8 @@ const std::string cilium_tls_http_proxy_config_fmt = R"EOF(
                   prefix: "/"
       transport_socket:
         name: "cilium.tls_wrapper"
+        typed_config:
+          "@type": type.googleapis.com/cilium.DownstreamTlsWrapperContext
```
[the remark this answers is not in the document]
Comment (sayboras) [reply tests/cilium_tls_http_integration_test.cc:130]: oh is it mapped automatically based on namespace (e.g. cilium) and type (e.g. DownstreamTlsWrapperContext)?

[the remark this answers is not in the document]
Comment (sayboras) [reply tests/cilium_tls_http_integration_test.cc:130]: basically, how to map between name (e.g cilium.tls_wrapper) and type (e.g. type.googleapis.com/cilium.DownstreamTlsWrapperContext)

[the remark this answers is not in the document]
Comment (jrajahalme) [reply tests/cilium_tls_http_integration_test.cc:130]: I think Envoy does this automatically via the definition of `createEmptyConfigProto()` that returns the type.

[the remark this answers is not in the document]
Comment (jrajahalme) [reply tests/cilium_tls_http_integration_test.cc:130]: The mapping is done by factory registration:
```
cilium/tls_wrapper.cc-REGISTER_FACTORY(UpstreamTlsWrapperFactory,
cilium/tls_wrapper.cc:                 Server::Configuration::UpstreamTransportSocketConfigFactory);
```

[the remark this answers is not in the document]
Comment (sayboras) [reply tests/cilium_tls_http_integration_test.cc:130]: thanks :100:

**Review by sayboras — APPROVED** [review]
:)

**Pull request merged**